theano/sandbox/gpuarray/elemwise.py

```diff
@@ -579,9 +595,14 @@ class GpuCAReduceCuda(HideC, CAReduceDtype):
 
     """
 
+    context_type = gpu_context_type
+
+    # This gets shadowed by the superclass versions
```

Comment:
Because the \__eq__ that \__props__ autodefines is in Op.  DimShuffle inherits from Op and redefines those methods, shadowing the ones from \__props__.

We should fix the \__props__ implementation so that it adds the methods to the new class rather than relying on inheritance.